py-polars/polars/internals/construction.py

```diff
@@ -905,18 +828,239 @@ def sequence_to_pydf(
+        schema_override = (
+            include_unknowns(schema_overrides, column_names) if schema_overrides else {}
+        )
+        if column_names and len(first_element) != len(column_names):
+            raise ShapeError("The row data does not match the number of columns")
+
+        for col, tp in schema_override.items():
+            if tp == Categorical:
+                schema_override[col] = Utf8
+
+        pydf = PyDataFrame.read_rows(
+            data,
+            infer_schema_length,
+            schema_override or None,
+        )
+        if column_names or schema_overrides:
+            pydf = _post_apply_columns(
+                pydf, column_names, schema_overrides=schema_overrides
+            )
+        return pydf
+
+    if orient == "col" or orient is None:
+        column_names, schema_overrides = _unpack_schema(
+            schema, schema_overrides=schema_overrides, n_expected=len(data)
+        )
+        data_series: list[PySeries] = [
+            pli.Series(
+                column_names[i], element, schema_overrides.get(column_names[i])
+            )._s
+            for i, element in enumerate(data)
+        ]
+        return PyDataFrame(data_series)
+
+    raise ValueError(
+        f"orient must be one of {{'col', 'row', None}}, got {orient} instead."
+    )
+
+
+@_sequence_to_pydf_dispatcher.register(tuple)
+def _sequence_of_tuple_to_pydf(
+    first_element: tuple[Any, ...],
+    data: Sequence[Any],
+    schema: SchemaDefinition | None,
+    schema_overrides: SchemaDict | None,
+    orient: Orientation | None,
+    infer_schema_length: int | None,
+) -> PyDataFrame:
+    if is_namedtuple(first_element):
         if schema is None:
-            column_names = []
+            schema = first_element._fields  # type: ignore[attr-defined]
+            if len(first_element.__annotations__) == len(schema):
+                schema = [
+                    (name, py_type_to_dtype(tp, raise_unmatched=False))
+                    for name, tp in first_element.__annotations__.items()
+                ]
+        elif orient is None:
+            orient = "row"
+
+    return _sequence_of_sequence_to_pydf(
+        first_element,
+        data=data,
+        schema=schema,
+        schema_overrides=schema_overrides,
+        orient=orient,
+        infer_schema_length=infer_schema_length,
+    )
+
+
+def _sequence_of_series_to_pydf(
+    first_element: pli.Series,
+    data: Sequence[Any],
+    schema: SchemaDefinition | None,
+    schema_overrides: SchemaDict | None,
+    orient: Orientation | None,
+    infer_schema_length: int | None,
+) -> PyDataFrame:
+    series_names = [s.name for s in data]
+    column_names, schema_overrides = _unpack_schema(
+        schema or series_names,
+        schema_overrides=schema_overrides,
+        n_expected=len(data),
+    )
+    data_series: list[PySeries] = []
+    for i, s in enumerate(data):
+        if not s.name:  # TODO: Replace by `if s.name is None` once allowed
```

Comment:
I think this comment can be removed as I think None for name of a Series is not planned anymore. `# TODO: Replace by `if s.name is None` once allowed` @ritchie46

Reply:
Yeah that's old, can be removed as far as I know.

Reply:
I just pushed a commit removing the comment